File: GeomPlanningSE2/1.cpp
#include <cmath>
#include <iostream>
#include <fstream>
#include <vector>
using namespace std;

int main()
{
  double grid[200][200];

  ifstream infile("/home/ljq/python test/111.txt");
  
  while (infile.good()&& !infile.eof())
  {
     for(int i=0;i<200;i++)
     {

         for(int j=0;j<200;j++)
         {             
             infile>>grid[i][j];                   
         }
     }
  }
  infile.close();
  
  ofstream outfile("/home/ljq/ompl_tutorial/OMPL_Sample/GeomPlanningSE2/33.txt");
       for(int i=0;i<200;i++)
     {

         for(int j=0;j<200;j++)
         {      
                outfile<<grid[i][j]<<" ";      
         
         }
         outfile<<"\n";
         
     }
  outfile.close();




        return 0;
}

File: GeomPlanningSE2/plot/1.py
from mpl_toolkits.mplot3d import Axes3D
import numpy
import matplotlib.pyplot as plt
data = numpy.loadtxt('path.dat')
#data1= numpy.loadtxt('/home/ljq/python test/111.txt')
data1= numpy.loadtxt('/home/ljq/ompl_tutorial/OMPL_Sample/GeomPlanningSE2/33.txt')
#data1= numpy.loadtxt('obstacle.dat')

x=[]
y=[]
fig = plt.figure()
for i in range(200):
        for j in range(200):
                if (data1[i][j]==0.0):
                        x.append(i)
                        y.append(j)
                        
                

#ax = fig.gca(projection='3d')
plt.plot(data[:,0],data[:,1],'.-')
plt.hold('on')
plt.grid('on')
#plt.fill(data1[:,0],data1[:,1],'r-')
plt.scatter(x,y)
plt.hold('on')
plt.grid('on')
plt.show()

